src/lib/Microsoft.Health.Extensions.Fhir.R4/Repository/FhirServiceRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir.Repository
+{
+    public class FhirServiceRepository : IFhirServiceRepository
+    {
+        private readonly IFhirClient _fhirClient;
+
+        private readonly CancellationTokenSource _cancellationTokenSource;
+
+        public FhirServiceRepository(IFhirClient fhirClient)
+        {
+            _fhirClient = EnsureArg.IsNotNull(fhirClient, nameof(fhirClient));
+            _cancellationTokenSource = new CancellationTokenSource();
+        }
+
+        public async Task<T> CreateResourceAsync<T>(
+            T resource,
+            string conditionalCreateCriteria = null,
+            string provenanceHeader = null,
+            CancellationToken cancellationToken = default)
+            where T : Resource
+        {
+            EnsureArg.IsNotNull(resource, nameof(resource));
+            _cancellationTokenSource.CancelAfter(60000);
+
+            CancellationToken token = cancellationToken.Equals(null) ? _cancellationTokenSource.Token : cancellationToken;
```

Comment:
>cancellationToken.Equals(null) ? _cancellationTokenSource.Token : cancellationToken; [](http://example.com/codeflow?start=38&length=84)

You want create linked tokens.  You can chain they together so you can have the timeout or the original ct that was passed in.  Right now you only have one.

Reply:
Thanks for suggesting this. Did some further reading and chained the internal and external tokens for each method together